### PR TITLE
Add RepoChangeSet

### PR DESCRIPTION
**Pull request by thieman**
@jsingle Specialized object for just doing this one thing.

### CODE REVIEW

File: dusty/changeset.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
+from .compiler.spec_assembler import get_repo_of_app_or_library, get_expanded_libs_specs
```
Comment (thieman) [inline dusty/changeset.py]: This kind of sucks but I don't think it's going to create any circular dependencies in practice

File: dusty/changeset.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        self.set_key = set_key
+        self.app_or_library_name = app_or_library_name
+        self.repos = {get_repo_of_app_or_library(self.app_or_library_name)}
+        for lib_name in self.primary_spec['depends']['libs']:
+            self.repos.add(get_repo_of_app_or_library(lib_name))
+
+    @property
+    def primary_spec(self):
+        expanded_specs = get_expanded_libs_specs()
+        if self.app_or_library_name in expanded_specs['apps']:
```
Comment (jsingle) [inline dusty/changeset.py]: we have a `get_app_or_lib` function

Comment (thieman) [inline dusty/changeset.py]: dear god where

Comment (thieman) [inline dusty/changeset.py]: oh we actually did the thing where the specs are a custom class?

File: dusty/changeset.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        expanded_specs = get_expanded_libs_specs()
+        return expanded_specs.get_app_or_lib(self.app_or_library_name)
+
+    def _get_current_sha_dict(self):
+        return {repo.remote_path: repo.local_commit_sha
+                for repo in self.repos}
+
+    def has_changed(self):
+        stored = get_config_value(constants.CONFIG_CHANGESET_KEY) or {}
+        return self._get_current_sha_dict() != stored.get(self.set_key, {})
```
Comment (jsingle) [inline dusty/changeset.py]: does dict comparison work like this? cool

**Comment by thieman** [comment]
Cool, merge it when you need it for your stuff

File: dusty/changeset.py (hunk 1 above)
Comment (jsingle) [inline dusty/changeset.py]: just PASS THE SPECS EVERYWHEREEEE

**Pull request merged**